### PR TITLE
Fix/15

### LINKED ISSUE #15: deposit balance checks
Provide feedback to user when they input more of a token than they have available

Eth will silently fail after clicking deposit:
Screen Shot 2021-11-29 at 10 25 13

ERC20s still try to submit the approval tx and fail:
Screen Shot 2021-11-25 at 11 31 21

### PR DESCRIPTION
**Pull request by aidan-starke**
Closes #15

### CODE REVIEW

File: utils/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { ethers } from "ethers";
+
+const ETH = "0x0000000000000000000000000000000000000000";
```
Comment (jordy25519) [inline utils/helpers.ts]: nit: recommend export here & import this value elsewhere vs. redefining it

File: utils/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  if (tokenAddress === ETH) {
+    balance = await provider.getBalance(account);
+  } else {
+    const token: ethers.Contract = new ethers.Contract(
+      tokenAddress,
+      minABI,
+      signer
+    );
+
+    balance = await token.balanceOf(account);
```
Comment (jordy25519) [inline utils/helpers.ts:22]: some erc20s e.g. USDC are not 18dp, will this be formatted with the right decimal places?

File: utils/helpers.ts [hunk 3]
```diff
@@ -1,33 +1,26 @@
...
       signer
     );
 
+    decimals = await token.decimals();
+
     balance = await token.balanceOf(account);
   }
 
-  return Number(ethers.utils.formatEther(balance));
+  return Number(ethers.utils.formatUnits(balance, decimals));
```
Comment (jordy25519) [inline utils/helpers.ts:25]: not sure how this format function behaves, if decimals is undefined (Eth case) will it fallback to 18?

**Review by jordy25519 — APPROVED** [review]
lgtm!

File: utils/helpers.ts (hunk 3 above)
Comment (aidan-starke) [reply utils/helpers.ts:25]: in ETH case we get balance from ethers provider 
`if (tokenAddress === ETH) {
    balance = await provider.getBalance(account);
  }`

**Pull request merged**